platform/storage/zephyr/nvs/src/mender-storage.c

```diff
@@ -50,15 +50,24 @@ static mender_err_t
 nvs_read_alloc(struct nvs_fs *nvs, uint16_t id, void **data, size_t *length) {
     ssize_t ret;
 
-    /* Retrieve length of the data */
-    ret = nvs_read(nvs, id, NULL, 0);
+    /* Allocate 1 byte fro the first peek read */
+    *data = malloc(1);
```

Comment:
This is pretty interesting. Do you know why we need to do it? I did not see any references to it while reviewing the NVS part.

Reply:
I believe it's a buggy implementation of the API on Lluis' target board/environment.

Reply:
The driver for my specific flash throws "invalid parameter" error when `data` is NULL, even when you ask to read 0 bytes. So yes, I consider it a bug in the driver, really.